src/react-native/18-maps/maps.md

```diff
@@ -2,25 +2,67 @@
 @parent learn-react-native 18
 @outline 3
 
-@description TODO
+@description Learn how to integrate Google Maps into your React Native application.
 
 @body
 
 ## Overview
 
 In this section, you will:
 
-- TODO
+- Integrate Google Maps
+- Be introduced to gradle
 
 ## Objective 1: Add Google Maps to Restaurant List page
 
-<img alt="Screenshot of the restaurant view with the title “Green Bay, Wisconsin.” There are two tabs at the top, List and Map, with Map selected. The map below is centered on Green Bay with nearby locations labeled. The bottom tab bar has icons for Place My Order and Settings." src="../static/img/react-native/18-maps/02-solution.png" style="max-height: 750px; border: 4px solid black; border-radius: 25px;"/>
+<img alt="Screenshot of the restaurant view with the title “Green Bay, Wisconsin.” There are two tabs at the top, List and Map, with Map selected. The map below is centered on Green Bay. The bottom tab bar has icons for Place My Order and Settings." src="../static/img/react-native/18-maps/01-solution.png" style="max-height: 750px; border: 4px solid black; border-radius: 25px;"/>
 
-TODO
+### React Native Maps
 
-### Concept TODO
+In this section, we will be using the `react-native-maps` library to integrate Google Maps into our application. This library provides several React Native components that can be used to render maps on both iOS and Android.
 
-TODO
+### Secrets Gradle Plugin
+
+Before we can use Google Maps in our application, we need to set up a Google Maps API key. We will be using the `secrets-gradle-plugin` to securely store our API key in our project.
+
+#### Gradle
+
+Gradle is a build automation tool that is used to compile and build the Android part of our React Native application. We can extend gradle's functionality by using plugins. In this case, we will be using the `secrets-gradle-plugin` to securely store our Google Maps API key. 
+
+#### Android Manifest
+
+The Android Manifest file is an XML file that contains important information about our application such as permissions, activities and services, configuration settings, and more.
+
+### MapView
+
+```tsx
+import MapView, { PROVIDER_GOOGLE } from "react-native-maps";
+
+function Map() {
+  return (
+    <MapView
+      style={{ minHeight: "100%", height: 500 }}
+      provider={PROVIDER_GOOGLE}
+      loadingEnabled
+      initialRegion={{
+        latitude: 37.78825,
+        longitude: -122.4324,
+        latitudeDelta: 0.0922,
+        longitudeDelta: 0.0421,
+      }}
+    ></MapView>
+  );
+}
+```
+
+The `MapView` component is the main component that we will be using to render the map. Because we are using a React Native library, we will later be able to pass child components to the `MapView` component to render `Markers` and other map elements.
+
+The component takes several props, but the most important ones are:
+
+- `style`: The style of the map.
```

Comment:
`minHeight` and `height` seem a little weird, could that be expanded on here? Why does a `height` need to be set if we have a `minHeight`?

Reply:
@snoattoh since i don't have the app running, can you take a look at this when you have a chance to confirm we can remove `minHeight` if we specify `height`? my gut instinct is that the weird behaviour here is due to parent component height being dynamic so the height of the `MapView` is being wonky (loading very small and not expanding correctly if not explicitly setting height)

I _think_ we can either remove `minHeight` and only specify height, _or_ make the wrapping parent component have a specific height value and then use `minHeight`?

Reply:
Yeah, as you've said we can remove the `minHeight` since we're providing a `height`.

Considering the `MapView` will throw an error if there is no `minHeight` even if the parent component provides a height, I don't feel very good about removing it even if we're providing a `height`, but I think as long as we explain this particular use case we can do whatever we prefer.